File: Python/Software Development Module - Python/Revision/Exam2014/Animals/AnimalApp.py
"""
AnimalApp.py
Billy Cussen
26/10/2020
"""

import Animal 
from Animal import myAnimal
import Cat 
from Cat import myCat
import Dog
from Dog import myDog

myAnimalApp = myAnimal()
print(myAnimalApp.getSpeak())
myAnimalApp = myCat()
print (myAnimalApp.getSpeak())
myAnimalApp = myDog()
print (myAnimalApp.getSpeak())

File: Java/Software Development Module - Java/Revision/Exam2014/AnimalApp.java
/*
AnimalApp.java
Billy Cussen 
07/08/2020
*/

import java.util.Scanner;

public class AnimalApp{

    public static void main(String [] args){

        //Variable 
        String animal = new String();
        String speak;
        boolean inputCheck;

        //Object
        Scanner keyboard;
        Animal myAnimal;

        //Input
        do{
            inputCheck = true;
            keyboard = new Scanner(System.in);

            System.out.print("Enter your Animal (Cat, Dog or Other): ");
            animal = keyboard.nextLine();

            if(!(animal.equalsIgnoreCase("Cat")||animal.equalsIgnoreCase("Dog")||animal.equalsIgnoreCase("Other"))){
                System.out.println("You need to enter Cat, Dog or Other!");
                inputCheck = false;
            }

            keyboard = null;

        }while(!inputCheck);

        if(animal.equalsIgnoreCase("Cat")){
            myAnimal = new Cat();
        } else if(animal.equalsIgnoreCase("Dog")){
            myAnimal = new Dog();
        } else{
            myAnimal = new Animal();
        }

        speak = myAnimal.speak();
        myAnimal = null;
        System.out.println(speak);

    }

}